kubernetes-addons/aws-node-termination-handler/main.tf

```diff
@@ -0,0 +1,182 @@
+      event_pattern = <<EOF
+{"source": ["aws.ec2"],"detail-type": ["EC2 Spot Instance Interruption Warning"]}
+EOF
+    },
+    {
+      name          = "NTHRebalanceRule",
+      event_pattern = <<EOF
+{"source": ["aws.ec2"],"detail-type": ["EC2 Instance Rebalance Recommendation"]}
+EOF
+    },
+    {
+      name          = "NTHInstanceStateChangeRule",
+      event_pattern = <<EOF
+{"source": ["aws.ec2"],"detail-type": ["EC2 Instance State-change Notification"]}
+EOF
+    },
+    {
+      name          = "NTHScheduledChangeRule",
+      event_pattern = <<EOF
+{"source": ["aws.health"],"detail-type": ["AWS Health Event"]}
+EOF
+    }
+  ]
+}
+
+resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_rule" {
+  count = length(local.rules)
+
+  name          = local.rules[count.index].name
+  event_pattern = local.rules[count.index].event_pattern
+}
+
+resource "aws_cloudwatch_event_target" "aws_node_termination_handler_rule_target" {
+  count = length(aws_cloudwatch_event_rule.aws_node_termination_handler_rule)
+
+  rule = aws_cloudwatch_event_rule.aws_node_termination_handler_rule[count.index].id
+  arn  = aws_sqs_queue.aws_node_termination_handler_queue.arn
+}
+
+module "irsa" {
+  source                     = "../irsa"
+  eks_cluster_name           = var.eks_cluster_name
+  kubernetes_namespace       = local.namespace
+  kubernetes_service_account = local.service_account_name
+  irsa_iam_policies          = [aws_iam_policy.aws_node_termination_handler_irsa.arn]
+}
+
+resource "aws_iam_policy" "aws_node_termination_handler_irsa" {
+  description = "IAM role policy for AWS Node Termination Handler"
+  name        = "${var.eks_cluster_name}-aws-nth-irsa"
+  policy      = data.aws_iam_policy_document.aws_node_termination_handler_irsa_policy_document.json
+}
+
+data "aws_iam_policy_document" "aws_node_termination_handler_irsa_policy_document" {
+  statement {
+    effect = "Allow"
+
+    resources = ["*"]
+
+    actions = [
+      "autoscaling:CompleteLifecycleAction",
+      "autoscaling:DescribeAutoScalingInstances",
+      "autoscaling:DescribeTags",
+      "ec2:DescribeInstances",
+      "sqs:DeleteMessage",
+      "sqs:ReceiveMessage"
+    ]
+  }
+}
+
+resource "helm_release" "aws_node_termination_handler" {
+  name       = "aws-node-termination-handler"
+  repository = "https://aws.github.io/eks-charts"
+  chart      = "aws-node-termination-handler"
+  version    = "0.16.0"
+  namespace  = local.namespace
+  verify     = false
+  timeout    = "1200"
+
+  set {
+    name  = "awsRegion"
+    value = data.aws_region.current.name
+  }
+  set {
+    name  = "serviceAccount.name"
+    value = local.service_account_name
+  }
+  set {
+    name  = "serviceAccount.create"
+    value = false
+  }
+  set {
+    name  = "enableSqsTerminationDraining"
+    value = true
+  }
+  set {
+    name  = "queueURL"
+    value = aws_sqs_queue.aws_node_termination_handler_queue.url
+  }
+  set {
+    name  = "enablePrometheusServer"
```

Comment:
Let's use the same pattern as other Helmcharts. Use default `values.yaml` file and `locals.tf` etc.
Please refer to KEDA addon

Reply:
will do